reconciler.go

```diff
@@ -86,14 +94,17 @@ func (r *Reconciler[T, PT]) Reconcile(ctx gocontext.Context, req ctrl.Request) (
 		return ctrl.Result{Requeue: true, RequeueAfter: 2 * time.Minute}, err
 	}
 
-	if err := r.Status().Update(r.DutyContext, obj); err != nil {
-		logger.Errorf("[kopper] failed to update status %s: %v", resourceName, err)
-		return ctrl.Result{Requeue: true, RequeueAfter: 2 * time.Minute}, err
+	if mgr, ok := any(obj).(StatusPatchGenerator); ok {
+		if patch := mgr.GenerateStatusPatch(original); patch != nil {
```

Comment:
Won't this break existing status patches ?

Reply:
There's https://github.com/flanksource/mission-control/pull/1802

Reply:
Shouldn't there be an else statement to make it backwards compatible ?